docs/docs/querying.md

```diff
@@ -284,22 +284,38 @@ something.findOne({
     // Will order by  otherfunction(`col1`, 12, 'lalala') DESC
     [sequelize.fn('otherfunction', sequelize.col('col1'), 12, 'lalala'), 'DESC'],
 
-    // Will order by name on an associated User
-    [User, 'name', 'DESC'],
+    // Will order an associated model's created_at using the model name as the association's name.
+    [Task, 'createdAt', 'DESC'],
 
-    // Will order by name on an associated User aliased as Friend
-    [{model: User, as: 'Friend'}, 'name', 'DESC'],
+    // Will order through an associated model's created_at using the model names as the associations' names.
+    [Task, Project, 'createdAt', 'DESC'],
 
-    // Will order by name on a nested associated Company of an associated User
-    [User, Company, 'name', 'DESC'],
+    // Will order by an associated model's created_at using the name of the association.
+    ['Task', 'createdAt', 'DESC'],
```

Comment:
@sushantdhiman I understand, you want to be able to sort by json property similar to how where clauses are constructed for next keys (http://docs.sequelizejs.com/en/latest/docs/querying/#jsonb)